### PR TITLE
utils: simplify mimalloc build handling

### PR DESCRIPTION
**Pull request by compnerd**
Use a bit more of a declarative style for describing the build rules for mimalloc.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
@swift-ci please smoke test

File: utils/build.ps1 [hunk 1]
```diff
@@ -1556,47 +1556,52 @@ function Build-Mimalloc() {
...
+    "lld-link.exe",
+    "ld.lld.exe",
+    "ld64.lld.exe"
+  )
+  foreach ($Tool in $Tools) {
+    $Binary = [IO.Path]::Combine($Dest, $Tool)
+    # Binary-patch in place
+    Invoke-Program "$SourceCache\mimalloc\bin\minject" @("-f", "-i", "-v", $Binary)
+    # Log the import table
+    Invoke-Program "$SourceCache\mimalloc\bin\minject" @("-I", $Binary)
```
Comment (hjyamauchi) [inline utils/build.ps1:1603]: The flag needs to be a lowercase L ("-l") instead of an uppercase I ("-I").

**Comment by compnerd** [comment]
@swift-ci please test Linux platform

**Comment by compnerd** [comment]
@swift-ci please smoke test

File: utils/build.ps1 (hunk 1 above)
Comment (compnerd) [reply utils/build.ps1:1603]: Whoops, thanks!

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Pull request merged**